### PR TITLE
fix(a11y): add ARIA dialog semantics and focus trap to ShareSheet modal

### LINKED ISSUE #2431: bug: ShareSheet modal lacks role="dialog" and aria-modal semantics — screen readers cannot identify it as a dialog
## Description

The `ShareSheet` component (`components/dashboard/ShareSheet.tsx`) renders a modal overlay but lacks proper ARIA dialog semantics. The overlay `<motion.div>` and inner panel have no `role="dialog"`, no `aria-modal="true"`, and no `aria-labelledby` pointing to the "Share Pulse" heading.

This means:
1. Screen readers do not announce it as a dialog when it opens
2. The overlay backdrop is not marked `aria-hidden="true"` — assistive tech may try to interact with it
3. There is no focus trap — Tab can escape to elements behind the overlay

**File:** `components/dashboard/ShareSheet.tsx` (lines ~170-200)

```tsx
// Current: plain motion.div with no dialog semantics
<motion.div
  id="share-sheet-overlay"
  ref={overlayRef}
  ...
  className="fixed inset-0 z-50 bg-black/60 ..."
>
  <motion.div ... className="relative w-full max-w-sm ...">
    {/* No role, no aria-modal */}
  </motion.div>
</motion.div>
```

## Steps to Reproduce

1. Go to any user dashboard (e.g., `/dashboard/octocat`)
2. Click the "Share" button on the ProfileCard
3. When the ShareSheet opens, use a screen reader (VoiceOver/NVDA)
4. Observe: no dialog announcement is made
5. Press Tab repeatedly — focus can escape behind the overlay

## Expected Behavior

- The inner panel should have `role="dialog"`, `aria-modal="true"`, and `aria-labelledby="share-sheet-title"`
- The heading "Share Pulse" should have `id="share-sheet-title"`
- Focus should be trapped within the panel while open
- Focus should move to the first interactive element when opened
- Focus should return to the trigger button when closed

## Screenshots / Logs

N/A — this is a screen reader / keyboard interaction issue.

## Suggested Fix

```tsx
<motion.div
  id="share-sheet-overlay"
  ref={overlayRef}
  onClick={onClose}
  className="fixed inset-0 z-50 bg-black/60 ..."
  aria-hidden="true"
>
</motion.div>
<motion.div
  role="dialog"
  aria-modal="true"
  aria-labelledby="share-sheet-title"
  className="relative w-full max-w-sm ..."
>
  <h2 id="share-sheet-title" ...>Share Pulse</h2>
  ...
</motion.div>
```

Add a focus trap using the same pattern as the Compare Profile modal fix.

### PR DESCRIPTION
**Pull request by nyxsky404**
## Description

Fixes #2431

The ShareSheet modal lacked proper ARIA dialog semantics, making it inaccessible to screen readers and keyboard-only users.

**Changes:**
- Added `role="dialog"`, `aria-modal="true"`, and `aria-labelledby="share-sheet-title"` to the dialog panel
- Added `id="share-sheet-title"` to the "Share Pulse" heading
- Separated the backdrop and dialog panel into siblings (backdrop gets `aria-hidden="true"`, dialog remains accessible)
- Added a focus trap (`handlePanelKeyDown`) that constrains Tab/Shift+Tab within the panel
- Focus trap handles the edge case where focus escapes the panel (forces it back inside)

All 16 existing ShareSheet tests continue to pass.

## Pillar

- [ ] �� Pillar 1 — New Theme Design
- [ ] 📐 Pillar 2 — Geometric SVG Improvement
- [ ] 🕐 Pillar 3 — Timezone Logic Optimization
- [x] 🛠️ Other (Bug fix, refactoring, docs)

## Visual Preview

N/A — this is a keyboard/screen reader accessibility fix. Visual appearance is unchanged.

## Checklist before requesting a review:

- [x] I have read the `CONTRIBUTING.md` file.
- [x] I have tested these changes locally (`localhost:3000/api/streak?user=YOUR_USERNAME`).
- [x] I have run `npm run format` and `npm run lint` locally and resolved all errors (CI will fail otherwise).
- [x] My commits follow the Conventional Commits format (e.g., `feat(themes): ...`, `fix(calculate): ...`).
- [x] I have updated `README.md` if I added a new theme or URL parameter.
- [x] I have started the repo.
- [x] I have made sure that i have only one commit to merge in this PR.
- [ ] The SVG output matches the CommitPulse "premium quality" aesthetic standard (no raw elements, smooth animations, correct fonts).
- [ ] (Recommended) I joined the CommitPulse Discord community for contributor discussions, mentorship, and faster PR support.

### CODE REVIEW

File: components/dashboard/ShareSheet.tsx [hunk 1]
```diff
@@ -114,6 +114,40 @@ endsolid commitpulse_monolith`;
     onClose();
   };
 
+  // Focus trap: constrain Tab navigation within the share sheet when open
+  const panelRef = useRef<HTMLDivElement>(null);
+  const handlePanelKeyDown = useCallback((e: React.KeyboardEvent) => {
```
[the remark this answers is not in the document]
Comment (nyxsky404) [reply components/dashboard/ShareSheet.tsx]: Fixed. Changed to `import { type KeyboardEvent as ReactKeyboardEvent } from 'react'` and updated the handler signature. The `React` namespace is no longer needed.

File: components/dashboard/ShareSheet.tsx [hunk 2]
```diff
@@ -114,6 +114,40 @@ endsolid commitpulse_monolith`;
...
+        e.preventDefault();
+        lastElement.focus();
+      }
+    } else {
+      if (activeEl === lastElement) {
+        e.preventDefault();
+        firstElement.focus();
+      }
+    }
+  }, []);
```
[the remark this answers is not in the document]
Comment (nyxsky404) [reply components/dashboard/ShareSheet.tsx]: Fixed. Added a `useEffect` that auto-focuses the first focusable element inside the panel when `isOpen` becomes true. This ensures focus moves into the dialog immediately on open, so the `onKeyDown` handler on the panel will receive Tab events. Combined with the existing Escape key listener on `window`, this provides a complete modal keyboard experience.

File: components/dashboard/ShareSheet.tsx [hunk 3]
```diff
@@ -276,100 +311,109 @@ endsolid commitpulse_monolith`;
...
+                                        ? 'JSON Downloaded!'
+                                        : opt.key === 'svg'
+                                          ? 'SVG Downloaded!'
+                                          : opt.key === 'stl'
+                                            ? 'STL Generated!'
+                                            : opt.label
+                            : state === 'error'
+                              ? 'Failed — try again'
+                              : opt.label}
+                        </span>
```
[the remark this answers is not in the document]
Comment (nyxsky404) [reply components/dashboard/ShareSheet.tsx]: Valid observation — the duplicated `png` branch and deeply nested ternary are pre-existing code that I didn't modify in this PR (the scope is adding ARIA/focus trap). Refactoring the success label logic into a lookup map would be a good follow-up PR to improve maintainability.

**Review by Aamod007 — APPROVED** [review]

**Review by Aamod007 — APPROVED** [review]
